docs/scripts/build-changelog.mjs

```diff
@@ -12,27 +13,68 @@ let changelogPath = await globby([
 
 console.log('📑 Generating changelogs...');
 
-const getReleaseInformation = async file => {
-  const log = await git.log({ file: path.resolve(file) });
+const getReleaseData = async () => {
```

Comment:
I asked Copilot to simplify this code snippet from l.16-77 and thats the result

```
const getReleaseData = async () => {
  const octokit = new Octokit({ auth: GITHUB_TOKEN });
  const response = await octokit.request('GET /repos/marigold-ui/marigold/releases', {
    headers: { 'X-GitHub-Api-Version': '2022-11-28' },
  });

  return response.data.reduce((releaseDates, release) => {
    const [, version] = release.name.match(/@([^@]+)$/) || [];
    if (!version) return releaseDates;

    if (!releaseDates.has(version)) {
      const releaseDate = new Date(release.published_at);
      const daysDifference = (new Date() - releaseDate) / (24 * 60 * 60 * 1000);
      releaseDates.set(version, {
        releaseDate,
        badge: daysDifference < 30 ? 'new' : undefined,
      });
    }
    return releaseDates;
  }, new Map());
};

const getReleaseInformation = (sourceText, releaseDates) => {
  const versionRegex = /## (\d{1,2}\.\d{1,2}\.\d{1,2})/gm;
  const versions = sourceText.match(versionRegex)?.map(v => v.replace('## ', '')) || [];

  if (!versions.length) {
    console.log('No versions found in the source text.');
    return [];
  }

  return versions.map(version => ({
    version,
    releaseDate: releaseDates.get(version)?.releaseDate || null,
    badge: releaseDates.get(version)?.badge || undefined,
  }));
```

Changes made:

- Combined the Map initialization and population in one step within reduce.
- Removed unnecessary intermediate variables.
- Simplified the calculation of daysDifference.
- Used optional chaining for safer access in getReleaseInformation.

Reply:
can you ask him to not use `.reduce`? :D I always find this makes the code harder to comprehend

Reply:
I gave this chatGPT 4o:


```js
const getReleaseData = async () => {
  const octokit = new Octokit({ auth: GITHUB_TOKEN });
  const response = await octokit.request('GET /repos/marigold-ui/marigold/releases', {
    headers: {
      'X-GitHub-Api-Version': '2022-11-28',
    },
  });

  const releases = response.data;
  const releaseDates = new Map();

  releases.forEach((release) => {
    const versionMatch = release.name.match(/@([^@]+)$/);
    const version = versionMatch ? versionMatch[1] : null;
    if (!version) return;

    if (!releaseDates.has(version)) {
      const releaseDate = new Date(release.published_at);
      const daysSinceRelease = calculateDaysSince(releaseDate);
      const badge = daysSinceRelease < 30 ? 'new' : undefined;

      releaseDates.set(version, {
        releaseDate,
        badge,
      });
    }
  });

  return releaseDates;
};

const calculateDaysSince = (date) => {
  const today = new Date();
  const timeDifference = today - date;
  const millisecondsInADay = 24 * 60 * 60 * 1000;
  return Math.round(timeDifference / millisecondsInADay);
};
```

### Changes Made:
1. **Extracted Regular Expression Match**: Stored the regex match result in `versionMatch` to simplify `version` extraction and improve readability.
2. **Renamed Variables for Clarity**:
   - Changed `aDayInMs` to `millisecondsInADay` for better understanding.
3. **Extracted Day Calculation Logic**: Moved the calculation of the number of days since the release into a separate function, `calculateDaysSince`, for clarity and reuse.
4. **Simplified Badge Logic**: Directly assigned `badge` during the calculation.

Reply:
This is exactly the same except you moved. the calculatoin of date difference out of it right?

Reply:
I changed it can you have a look again?